Add libera.chat to the list of available networks and make it the
default.

Patches copied from upstream source repository:

https://github.com/hexchat/hexchat/commit/a25f2381689d2c2279a0e43b33f6c0ec8305a096
https://github.com/hexchat/hexchat/commit/d3545f37cd5f551ed8bc0ab7b20e5c8140adc0a6

From a25f2381689d2c2279a0e43b33f6c0ec8305a096 Mon Sep 17 00:00:00 2001
From: Panagiotis Vasilopoulos <hello@alwayslivid.com>
Date: Wed, 19 May 2021 16:25:16 +0300
Subject: [PATCH] Add Libera Chat to network list

---
 src/common/servlist.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/common/servlist.c b/src/common/servlist.c
index e44a3f2df..33bd80f2d 100644
--- a/src/common/servlist.c
+++ b/src/common/servlist.c
@@ -239,6 +239,9 @@ static const struct defaultserver def[] =
 	/* Self signed */
 	{0,			"irc.librairc.net"},
 
+	{"Libera Chat", 0, 0, 0, LOGIN_SASL, 0, TRUE},
+	{0,			"irc.libera.chat"},
+
 #ifdef USE_OPENSSL
 	{"LinkNet",	0},
 	{0,			"irc.link-net.org/+7000"},
From d3545f37cd5f551ed8bc0ab7b20e5c8140adc0a6 Mon Sep 17 00:00:00 2001
From: Patrick Griffis <pgriffis@igalia.com>
Date: Sun, 23 May 2021 21:15:43 -0500
Subject: [PATCH] Change default network to Libera.Chat

---
 src/common/servlist.c | 4 ++--
 src/fe-gtk/joind.c    | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/common/servlist.c b/src/common/servlist.c
index de75c8b9a..93557f975 100644
--- a/src/common/servlist.c
+++ b/src/common/servlist.c
@@ -240,7 +240,7 @@ static const struct defaultserver def[] =
 	/* Self signed */
 	{0,			"irc.librairc.net"},
 
-	{"Libera Chat", 0, 0, 0, LOGIN_SASL, 0, TRUE},
+	{"Libera.Chat", 0, 0, 0, LOGIN_SASL, 0, TRUE},
 	{0,			"irc.libera.chat"},
 
 #ifdef USE_OPENSSL
@@ -942,7 +942,7 @@ servlist_load_defaults (void)
 {
 	int i = 0, j = 0;
 	ircnet *net = NULL;
-	guint def_hash = g_str_hash ("freenode");
+	guint def_hash = g_str_hash ("Libera.Chat");
 
 	while (1)
 	{
diff --git a/src/fe-gtk/joind.c b/src/fe-gtk/joind.c
index f1d3da504..ce3cbcaec 100644
--- a/src/fe-gtk/joind.c
+++ b/src/fe-gtk/joind.c
@@ -247,7 +247,7 @@ joind_show_dialog (server *serv)
 							G_CALLBACK (joind_ok_cb), serv);
 							
 	if (serv->network)
-		if (g_ascii_strcasecmp(((ircnet*)serv->network)->name, "freenode") == 0)
+		if (g_ascii_strcasecmp(((ircnet*)serv->network)->name, "Libera.Chat") == 0)
 		{
 			gtk_entry_set_text (GTK_ENTRY (entry1), "#hexchat");
 		}
